### PR TITLE
Test cases for parsing "missing"

### PR DESCRIPTION
**Pull request by sjakobi**
The Haskell implementation currently gets this wrong. See
https://github.com/dhall-lang/dhall-haskell/issues/1454.

### CODE REVIEW

File: tests/parser/success/preferMissingNoSpacesA.dhall [hunk 1]
```diff
@@ -0,0 +1 @@
+missing//foo
```
Comment (Gabriel439) [inline tests/parser/success/preferMissingNoSpacesA.dhall]: This is a really evil test (in a good way) 🙂

File: tests/parser/success/missingSlashA.dhall [hunk 2]
```diff
@@ -0,0 +1 @@
+missing/
```
Comment (sjakobi) [inline tests/parser/success/missingSlashA.dhall]: Should I add another test for the variable `missing//`? :thinking:

Comment (sjakobi) [reply tests/parser/success/missingSlashA.dhall]: …and the prefer-expression `missing// foo`…

Comment (Gabriel439) [reply tests/parser/success/missingSlashA.dhall]: I think the `missing//foo` test is good enough

**Comment by Nadrieril** [comment]
Hmm, my parser parses `missing//foo` as a variable named `"missing//foo"`. Since slashes are allowed in variable names, I feel like this might be the correct parse ?
I'm reasonably confident in my parser since it's derived mostly mechanically from the abnf. If that's not the expected parse, I think that means either that the abnf is ambiguous or that it needs to be updated.

**Comment by sjakobi** [comment]
Hmm, I admittedly wasn't very confident about that test case – I hoped the reviewers would catch it if it's wrong…

**Comment by philandstuff** [comment]
@Nadrieril my reading of the ABNF is that the test in this PR is correct and you are wrong.

The ABNF says to prefer as many repetitions in a repeating construct as possible.  The rule for  `prefer-expression` should have as many repeats of `whsp prefer whsp combine-types-expression` as possible.  Therefore, you need to parse `missing//foo` as a `prefer-expression`, consisting of the `combine-types-expression` `missing` applied to the `combine-types-expression` `foo`; any other parse fails to parse as many of this construct as possible.

**Pull request merged**